SU2_CFD/src/iteration_structure.cpp

```diff
@@ -342,7 +342,7 @@ void CIteration::SetMesh_Deformation(CGeometry **geometry,
   /*--- Perform the elasticity mesh movement ---*/
   if (config->GetDeform_Mesh()) {
 
-    if(kind_recording != MESH_DEFORM){
+    if ((kind_recording != MESH_DEFORM) && !config->GetMultizone_Problem()) {
       /*--- In a primal run, AD::TapeActive returns a false ---*/
```

Comment:
Mesh deformation needs to be recorded in FSI problems also for the conservative variables as it establishes the connection between structural displacements and fluid variables.